catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/CacheKey.java

```diff
@@ -13,50 +13,67 @@
  */
 package ddf.catalog.cache.impl;
 
+import java.util.Collections;
 import java.util.Set;
 
+import org.apache.commons.lang.Validate;
+
 import ddf.catalog.content.data.ContentItem;
 import ddf.catalog.data.Metacard;
 import ddf.catalog.operation.ResourceRequest;
 
+/**
+ * Class used to create keys for the {@link ResourceCache} class.
+ */
 public class CacheKey {
 
     private Metacard metacard;
 
+    private Set<String> propertyNames = Collections.emptySet();
+
     private ResourceRequest resourceRequest;
 
+    /**
+     * Constructor used for keys generated from a {@link Metacard} and {@link ResourceRequest}.
+     *
+     * @param metacard        metacard to use to generate the key
+     * @param resourceRequest resource request object to use to generate the key
+     */
     public CacheKey(Metacard metacard, ResourceRequest resourceRequest) {
+        this(metacard);
 
-        if (metacard == null) {
-            throw new IllegalArgumentException("Metacard must not be null.");
-        }
+        Validate.notNull(resourceRequest, "ResourceRequest must not be null.");
 
-        if (resourceRequest == null) {
-            throw new IllegalArgumentException("ResourceRequest must not be null.");
-        }
+        this.resourceRequest = resourceRequest;
+        this.propertyNames = resourceRequest.getPropertyNames();
```

Comment:
Nothing prevents `resourceRequest.getPropertyNames()` from being null which would cause an NPE in your `generateKey()` method.